src/main/java/com/mixpanel/android/util/HttpService.java

```diff
@@ -27,8 +27,36 @@
  * An HTTP utility class for internal use in the Mixpanel library. Not thread-safe.
  */
 public class HttpService implements RemoteService {
+
+    private static boolean sAdBlockerEnabled;
+
+    @Override
+    public void checkAdBlockerEnabled() {
+        Thread t = new Thread(new Runnable() {
+            public void run() {
+                try {
+                    InetAddress apiMixpanelInet = InetAddress.getByName("api.mixpanel.com");
+                    InetAddress decideMixpanelInet = InetAddress.getByName("decide.mixpanel.com");
```

Comment:
That's the whole point of this PR. We need to check whether we are able to communicate with Mixpanel or not.